api/v1beta1/flinkcluster_types.go

```diff
@@ -487,40 +525,42 @@ type BatchSchedulerSpec struct {
 
 // FlinkClusterSpec defines the desired state of FlinkCluster
 type FlinkClusterSpec struct {
-	// The version of Flink to be managed. This version must match the version in the image.
+	// _(Required)_ The version of Flink to be managed. This version must match the version in the image.
 	FlinkVersion string `json:"flinkVersion"`
 
-	// Flink image spec for the cluster's components.
+	// _(Required)_ Flink image for JobManager, TaskManager and job containers.
 	Image ImageSpec `json:"image"`
 
 	// The service account assigned to JobManager, TaskManager and Job submitter Pods. If empty, the default service account in the namespace will be used.
 	ServiceAccountName *string `json:"serviceAccountName,omitempty"`
 
-	// Deprecated: BatchSchedulerName specifies the batch scheduler name for JobManager, TaskManager.
+	// _*(Deprecated)*_ BatchSchedulerName specifies the batch scheduler name for JobManager, TaskManager.
 	// If empty, no batch scheduling is enabled.
 	BatchSchedulerName *string `json:"batchSchedulerName,omitempty"`
 
 	// BatchScheduler specifies the batch scheduler for JobManager, TaskManager.
 	// If empty, no batch scheduling is enabled.
 	BatchScheduler *BatchSchedulerSpec `json:"batchScheduler,omitempty"`
 
-	// Flink JobManager spec.
+	// _(Required)_ Flink JobManager spec.
 	JobManager JobManagerSpec `json:"jobManager"`
 
-	// Flink TaskManager spec.
+	// _(Required)_ Flink TaskManager spec.
 	TaskManager TaskManagerSpec `json:"taskManager"`
 
-	// (Optional) Job spec. If specified, this cluster is an ephemeral Job
```

Comment:
we can keep `Optional` now ... it's easy to maintain now

Reply:
Can I add `optional` mark into all the omitted fields except to fill the default value?
These include not only job, taskmanager.port but volume, volumemount, initcontainers, and so on..

Or is there something you are thinking about like excluding the k8s resource fields?

Reply:
Yeah we can add the `optional` mark to everything that is really optional.

Reply:
I have updated it~!